etc/profile-m-z/steam.profile

```diff
@@ -42,17 +52,27 @@ include disable-interpreters.inc
 include disable-passwdmgr.inc
 include disable-programs.inc
 
+mkdir ${HOME}/.config/Epic
+mkdir ${HOME}/.config/Loop_Hero
+mkdir ${HOME}/.config/ModTheSpire
+mkdir ${HOME}/.config/RogueLegacy
 mkdir ${HOME}/.config/unity3d
 mkdir ${HOME}/.killingfloor
+mkdir ${HOME}/.klei
 mkdir ${HOME}/.local/share/3909/PapersPlease
 mkdir ${HOME}/.local/share/aspyr-media
 mkdir ${HOME}/.local/share/bohemiainteractive
 mkdir ${HOME}/.local/share/cdprojektred
+mkdir ${HOME}/.local/share/Dredmor
 mkdir ${HOME}/.local/share/FasterThanLight
 mkdir ${HOME}/.local/share/feral-interactive
 mkdir ${HOME}/.local/share/IntoTheBreach
 mkdir ${HOME}/.local/share/Paradox Interactive
+mkdir ${HOME}/.local/share/PillarsOfEternity
+mkdir ${HOME}/.local/share/RogueLegacy*
```

Comment:
@davidebeatrici this one? Globbing on `mkdir` does not make sense.

@kmk3 Catching of globbing chars on `mkdir`/`mkfile` would make sense in your linter scripts.

Reply:
Yep, this one.

Reply:
> @kmk3 Catching of globbing chars on `mkdir`/`mkfile` would make sense in your
> linter scripts.

That's true; thanks for the reminder.  Link for cross-reference:
<https://github.com/netblue30/firejail/issues/3979#issuecomment-778491722>

Sorry for not following up on that earlier; some things got in the way IRL and
firejail moves quite fast so I've been doing mostly just simple/self-contained
code changes for a while.

I'll try to rebase my makefile/linter-related branches and create PRs for them
by the end of next week.

Reply:
Should we only drop the start or the complete line?

Reply:
>     noblacklist ${HOME}/.local/share/RogueLegacy*
>     [...]
>     mkdir ${HOME}/.local/share/RogueLegacy*
>     [...]
>     whitelist ${HOME}/.local/share/RogueLegacy*

@matthew-cline Why was globbing used here?  According to PCGamingWiki, [Rogue
Legacy only uses a single path on `$XDG_DATA_HOME`][1]:

* Configuration: `$XDG_CONFIG_HOME/RogueLegacy/`
* Data: `$XDG_DATA_HOME/RogueLegacy/`

And Rogue Legacy 2 [is non-native and has spaces on its paths][2]:

* Configuration (Windows): `HKEY_CURRENT_USER\Software\Cellar Door Games\Rogue Legacy 2`
* Data (Windows): `%USERPROFILE%\AppData\LocalLow\Cellar Door Games\Rogue Legacy 2`

Do you know of any other paths?

[1]: https://www.pcgamingwiki.com/wiki/Rogue_Legacy#Game_data
[2]: https://www.pcgamingwiki.com/wiki/Rogue_Legacy_2#Game_data

Reply:
> Do you know of any other paths?

Rogue Legacy 1 created the directory `$XDG_DTA_HOME/RogueLegacyStorageContainer/` in addition to `$XDG_DATA_HOME/RogueLegacy/`, which is why I used globbing.

Reply:
> > Do you know of any other paths?
> 
> Rogue Legacy 1 created the directory
> `$XDG_DTA_HOME/RogueLegacyStorageContainer/` in addition to
> `$XDG_DATA_HOME/RogueLegacy/`, which is why I used globbing.

Interesting; which version of Rogue Legacy do you use?

It seems that the former is the legacy path:

From <https://steamcommunity.com/app/241600/discussions/1/846957366713233279/>:

Rogue Legacy > Technical Support > Topic Details

Cellar Door Games  [developer]	27 Jun, 2013 @ 11:58am:

> Where are my crash logs?

[...]

> WHERE CAN I FIND MY SAVE FILES?
> 
> We figured a lot of people would be asking this question. Your save files can
> be located here:
> 
> Windows %USERPROFILE%\My Documents\SavedGames\RogueLegacy
> OS X ~/Library/Application Support/RogueLegacyStorageContainer/
> Linux ~/.local/share/RogueLegacyStorageContainer/
>
>
> WHERE CAN I FIND MY CONFIG FILE?
> 
> If you want to manually change game config data (like fullscreen or
> windowed), you can do so by opening the GameConfig.ini file with notepad. The
> config file can be located here:
> 
> Windows %APPDATA%\RogueLegacy
> OS X ~/Library/Application Support/RogueLegacyStorageContainer/
> Linux ~/.config/RogueLegacyStorageContainer/
>
> Last edited by Cellar Door Games; 21 Jan, 2014 @ 6:04pm

From <https://gaming.stackexchange.com/questions/121980/where-is-the-save-file-for-rogue-legacy-located/142452#142452>:

@ryanpcmcquen answered Nov 17 '13 at 23:33:

> Or for Linux:
>
>     ~/.local/share/RogueLegacyStorageContainer/

[Sergio][1] Aug 20 '16 at 14:09:

> Mine was in
> `~/.local/share/RogueLegacy/RogueLegacyStorageContainer/AllPlayers`, but
> wouldn't know without this answer. Thanks!

@ryanpcmcquen Aug 21 '16 at 16:36:

> Ah, they may have moved it in newer versions. Thanks!

From <https://gaming.stackexchange.com/questions/147008/how-to-run-rogue-legacy-for-linux-on-a-specific-monitor>:

@l0b0 asked Dec 19 '13 at 22:33:

[...]

> I had a look in `~/.config/RogueLegacyStorageContainer/GameConfig.ini`

So it seems that there are at least 4 paths that Rogue Legacy may search:

* `~/.config/RogueLegacy`
* `~/.config/RogueLegacyStorageContainer`
* `~/.local/share/RogueLegacy`
* `~/.local/share/RogueLegacyStorageContainer`

I'll create a PR to use them instead of the globbing.

---

P.S. to the people mentioned: I figured that you might find interesting where
your SE answers/etc end up being useful outside of SE.  If this is too spammy,
please let me know.

[1]: https://gaming.stackexchange.com/users/27172/sergio

Reply:
> I'll create a PR to use them instead of the globbing.

To expand a bit, I think doing so is more useful in this case, as:

* The paths are known a priori (unlike, say, `/var/lib/libpcre*`)
* There aren't too many of them
* It gets in the way of whitelisting (due to mkfile/mkdir)

Reply:
> Interesting; which version of Rogue Legacy do you use?

I'm using the current version, but I years ago I played the older version and I guess I never removed the old paths.

Reply:
> I'll create a PR to use them instead of the globbing.

Actually, I had forgotten that path clobbering + whitelisting is an issue.  In
that case, just creating all dirs might cause problems.  I've opened PR #4225,
which has further details on that.  Let's continue the discussion over there.

@matthew-cline yesterday:

> > Interesting; which version of Rogue Legacy do you use?
> 
> I'm using the current version, but I years ago I played the older version and
> I guess I never removed the old paths.

Nice.  I wonder which paths are used for my config/data.  The saves are from
years ago as well, but unfortunately I can't easily access them at the moment.